### ISSUE TITLE
Storybook: Make sure that subcomponents are properly documented

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#53541]
Storybook 7 (#53520) does not support the display of subcomponents in the props tables, so they had to be removed.

We should go through the affected components and make sure that they are appropriately documented in some way or another. Simple subcomponents may be sufficiently documented just as a story of the parent component (e.g. [`DropdownContentWrapper`](https://wordpress.github.io/gutenberg/?path=/docs/components-dropdown--with-more-padding)). More complex subcomponents may warrant their own story pages. Maybe there are other ways as well.

## Affected subcomponents

- [ ] AlignmentMatrixControl.Icon
- [ ] CardHeader, CardBody, CardDivider, CardMedia, CardFooter
- [ ] CircularOptionPicker.Option, CircularOptionPicker.ButtonAction, CircularOptionPicker.DropdownLinkAction
- [ ] DropdownMenuItem, DropdownSubMenu, DropdownSubMenuTrigger, DropdownMenuSeparator, DropdownMenuCheckboxItem, DropdownMenuGroup, DropdownMenuLabel, DropdownMenuRadioGroup, DropdownMenuRadioItem
- [x] DropdownContentWrapper
- [ ] FlexBlock, FlexItem
- [ ] InputControlPrefixWrapper, InputControlSuffixWrapper
- [ ] Item (in ItemGroup)
- [ ] NavigationBackButton, NavigationGroup, NavigationItem, NavigationMenu
- [ ] NavigatorScreen, NavigatorButton, NavigatorBackButton
- [ ] NoticeList
- [ ] PanelRow, PanelBody
- [ ] Radio
- [ ] Fill, SlotFillProvider
- [ ] ToggleGroupControlOption, ToggleGroupControlOptionIcon
- [ ] ToolbarButton, ToolbarGroup, ToolbarItem, ToolbarDropdownMenu
- [ ] ToolsPanelItem
- [ ] TreeGridRow, TreeGridCell

**Comment by bangank36** [comment]
If the subcomponents are included as the parent's story, then it won't make sense if someone visits its story page, eg:[AlignmentMatrix.Icon](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-alignmentmatrixcontrol--icon-subcomponent) has the main component controls only

> More complex subcomponents may warrant their own story pages

I think this suggestion make sense, and the sidebar layout can be like
<img src="https://user-images.githubusercontent.com/321738/111605030-3e207e00-87d6-11eb-8802-9b0b6499fb0a.png">

Another approach is customize the `docs.page` prop to create a custom docs page with `argsTable` and bring back the `subcomponents` 
[@psui preview.tsx](https://github.com/x7ddf74479jn5/psui/blob/4e8e850dc5897e698a5da4c8475662a085de77ff/packages/components/.storybook/preview.tsx#L42) / [HoverCard stories](https://x7ddf74479jn5.github.io/psui/?path=/docs/components-hovercard--docs)

**Comment by mirka** [comment]
> > More complex subcomponents may warrant their own story pages
> 
> I think this suggestion make sense, and the sidebar layout can be like

Yes, I think this is the way to go.

> Another approach is customize the `docs.page` prop to create a custom docs page with `argsTable` and bring back the `subcomponents` [@psui preview.tsx](https://github.com/x7ddf74479jn5/psui/blob/4e8e850dc5897e698a5da4c8475662a085de77ff/packages/components/.storybook/preview.tsx#L42) / [HoverCard stories](https://x7ddf74479jn5.github.io/psui/?path=/docs/components-hovercard--docs)

I wasn't aware of this thing, thanks. I tried it out in #53751 but realized it would remove the Controls from the Docs pages. I'm a bit hesitant to go through with this now, because I feel like it would be worse for the overall developer experience 😅

**Comment by bangank36** [comment]
> but realized it would remove the Controls from the Docs pages

FYI, you will need the prop `story` for ArgsTable to show controls for the main story, you can explicitly set its value as 
```
<ArgsTable story="^" />
```

or import PRIMARY_STORY from `addon-docs`
```
{
     PRIMARY_STORY
 } from '@storybook/addon-docs';
<ArgsTable story={PRIMARY_STORY} />
```

Found similar usage in [equinor/design-system](https://github.com/equinor/design-system), they used a [PropsTable](https://github.com/equinor/design-system/blob/dc8c22fcf950bab923ed11f9432bdb76e7891924/packages/eds-lab-react/.storybook/components/PropsTable.tsx#L3) in mdx stories. Example [Navigation-tabs stories](https://storybook.eds.equinor.com/?path=/docs/navigation-tabs--docs)

Recording
![chrome-capture-2023-7-17](https://github.com/WordPress/gutenberg/assets/10071857/c80ee4ff-817b-4e59-ab4c-f339010e5a66)

**Comment by mirka** [comment]
@bangank36 It worked 😱 #53751

**Comment by bangank36** [comment]
> It worked

💯 
Personally, I think it is hard to find the full list of support props for this component 😓
PS: I think you meant `ArgsTable` not `ArgsTypes`? https://github.com/WordPress/gutenberg/pull/53751#issue-1853999227

**Comment by mirka** [comment]
> PS: I think you meant `ArgsTable` not `ArgsTypes`? [#53751 (comment)](https://github.com/WordPress/gutenberg/pull/53751#issue-1853999227)

You're right, I was just reading the v6 to v7 migration docs and got them mixed up 🤦

**Comment by mirka** [comment]
Closing because subcomponent support in the props tables were "[undeprecated](https://github.com/storybookjs/storybook/issues/20782#issuecomment-1892178858)" 😄
